spec/rspec/mocks/verifying_doubles/class_double_with_class_loaded_spec.rb

```diff
@@ -165,7 +165,7 @@ class << self
 
         context "on a class with a private `new`" do
           it 'uses the method signature from `#initialize` for arg verification' do
-            if RSpec::Support::Ruby.jruby? && RSpec::Support::Ruby.jruby_version < '9.2.1.0'
+            if RSpec::Support::Ruby.jruby? && JRUBY_VERSION < '9.2.1.0'
```

Comment:
Revert

Reply:
I see that a naïve string comparison is inferior to our `rspec-support`'s implementation of `jruby_version`, e.g.:
```ruby
'3.2' < '3.10' # => false
'2.3.rc1' < '2.3' # => false
```

However, here, in `rspec-mocks` we exclusively compare JRuby version in specs.
And in both cases, we compare with `'9.2.1.0'`.
I could not figure out a version number that RVM would install (and it only provides stable versions) that would be incorrectly compared. Can you?

I understand your concern, however, and I will revert it in `rspec-support`
On the one hand, it would be safe for future changes to keep the correct version comparison.
On the other hand, I remember you mentioned that we're aiming to drop specific Ruby implementation support, and "support Rubies that are MRI-compatible".

Just for the reference, this is a comprehensive list of our JRuby version comparisons. We only compare to "safe to compare" `9.2.0.0` and `9.2.1.0`.
```
$ rg JRUBY_VERSION
spec/integration/suite_hooks_errors_spec.rb
11:  if RSpec::Support::Ruby.jruby? && JRUBY_VERSION > '9.2.0.0'
99:      if RSpec::Support::Ruby.jruby? && JRUBY_VERSION > '9.2.0.0'

spec/rspec/core/formatters/exception_presenter_spec.rb
509:            if RSpec::Support::Ruby.jruby? && JRUBY_VERSION < '9.2.0.0'
526:            if RSpec::Support::Ruby.jruby? && JRUBY_VERSION < '9.2.0.0'

spec/rspec/support/ruby_features_spec.rb
125:          if Ruby.jruby? && JRUBY_VERSION < '9.2.1.0'

lib/rspec/support/ruby_features.rb
68:      if Ruby.rbx? || (Ruby.jruby? && JRUBY_VERSION < '9.2.1.0')

spec/rspec/expectations/failure_aggregator_spec.rb
70:        pending "This is broken on 9.2.x.x" unless JRUBY_VERSION < '9.2.0.0'

lib/rspec/expectations/failure_aggregator.rb
55:      if RSpec::Support::Ruby.jruby? && JRUBY_VERSION < '9.2.0.0'

spec/rspec/mocks/partial_double_spec.rb
596:            if RSpec::Support::Ruby.jruby? && JRUBY_VERSION < '9.2.1.0'

spec/rspec/mocks/verifying_doubles/class_double_with_class_loaded_spec.rb
168:            if RSpec::Support::Ruby.jruby? && JRUBY_VERSION < '9.2.1.0'
```

Please do not get me wrong, I do not resist bringing it back, but I'd like to know why exactly we're keeping it.